src/lib.rs

```diff
@@ -267,9 +266,10 @@ extern "C" fn init(callbacks: *mut PluginCallbacks, config_path: *const c_char)
                 message_threads
             };
 
+            let mut senders = Vec::with_capacity(num_threads);
```

Comment:
Creating `Vec` with `Vec::with_capacity` is a best practice when you know the size in advance.
From memory allocation point of view, I read there is no allocation when using `Vec::new`, there is an allocation for 4 items on the first push, then a new allocation with capacity x2 when the capacity is reached and so on.
Here nothing changes fundamentally, this code is executed when the plugin initialize, we may save an extra allocation if you have more than 4 cpus. :)